--- b/src/mysql_visual_explain_cli/graphics/cairo_utils.py
+++ a/src/mysql_visual_explain_cli/graphics/cairo_utils.py
@@ -20,7 +20,8 @@
 # along with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA

+import cairocffi
+from cairocffi import cairo
-import cairo
 from math import ceil, floor, fabs, atan, pi


@@ -32,7 +34,7 @@
         return cairo.cairo_surface_status(self.s)

     def write_to_png(self, file):
+        cairo.cairo_surface_write_to_png(self.s, file.encode())
-        cairo.cairo_surface_write_to_png(self.s, file)


 class ImageSurface(Surface):
@@ -126,10 +128,10 @@
         cairo.cairo_restore(self.cr)
  
     def set_dash(self, dashes, offset):
+        cairo.cairo_set_dash(self.cr, dashes, len(dashes), offset)
-        cairo.cairo_set_dash(self.cr, dashes, offset)

     def set_font(self, family, italic=False, bold=False):
+        cairo.cairo_select_font_face(self.cr, family.encode(), cairo.CAIRO_FONT_SLANT_ITALIC if italic else cairo.CAIRO_FONT_SLANT_NORMAL, cairo.CAIRO_FONT_WEIGHT_BOLD if bold else cairo.CAIRO_FONT_WEIGHT_NORMAL)
-        cairo.cairo_select_font_face(self.cr, family, cairo.CAIRO_FONT_SLANT_ITALIC if italic else cairo.CAIRO_FONT_SLANT_NORMAL, cairo.CAIRO_FONT_WEIGHT_BOLD if bold else cairo.CAIRO_FONT_WEIGHT_NORMAL)

     def set_font_size(self, size):
         cairo.cairo_set_font_size(self.cr, size)
@@ -156,7 +158,7 @@
         cairo.cairo_mask_surface(self.cr, surface.s, x, y)

     def show_text(self, text):
+        cairo.cairo_show_text(self.cr, text.encode())
-        cairo.cairo_show_text(self.cr, text)

     def rounded_rect(self, x, y, w, h, r):
         self.move_to(x+r, y)
@@ -203,7 +205,9 @@
         cairo.cairo_stroke_preserve(self.cr)

     def text_extents(self, text):
+        extents = cairocffi.ffi.new('cairo_text_extents_t *')
+        cairo.cairo_text_extents(self.cr, text.encode(), extents)
+        return extents
-        return cairo.cairo_text_extents(self.cr, text)



--- b/src/mysql_visual_explain_cli/query_analysis/explain_renderer.py
+++ a/src/mysql_visual_explain_cli/query_analysis/explain_renderer.py
@@ -20,15 +20,11 @@
 # along with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA

-import mforms
-
-
-from workbench.log import log_error
+from logging import exception as log_error

-
-from workbench.graphics.canvas import VBoxFigure, Canvas, DiamondShapeFigure, RectangleShapeFigure, TextFigure, HFill, draw_varrow, draw_harrow
-from workbench.graphics.cairo_utils import ImageSurface, Context
-import cairo
+from mysql_visual_explain_cli.graphics.canvas import VBoxFigure, Canvas, DiamondShapeFigure, RectangleShapeFigure, TextFigure, HFill, draw_varrow, draw_harrow
+from mysql_visual_explain_cli.graphics.cairo_utils import ImageSurface, Context
+from cairocffi import cairo

 import io
 import json
@@ -426,9 +424,7 @@

     @property
     def rows_count(self):
+        return self.child_below.rows_produced
-        if self._context.server_version.is_supported_mysql_version_at_least(5, 7):
-            return self.child_below.rows_produced
-        return None


     def get_read_eval_cost(self):
